src/RamTransforms.cpp

```diff
@@ -427,6 +644,169 @@ bool MakeIndexTransformer::makeIndex(RamProgram& program) {
+                                std::make_unique<RamTupleElement>(indexOperation->getTupleId(), i),
+                                std::unique_ptr<RamExpression>(pattern.second[i]->clone()));
+                        condition = addCondition(std::move(condition), upperBound->clone());
+                    }
+
+                    // reset the bounds
+                    updatedPattern.first[i] = std::make_unique<RamUndefValue>();
+                    updatedPattern.second[i] = std::make_unique<RamUndefValue>();
+                }
+
+                if (condition) {
+                    auto nestedOp = std::unique_ptr<RamOperation>(indexOperation->getOperation().clone());
+                    auto filter = std::make_unique<RamFilter>(std::move(condition), std::move(nestedOp));
+
+                    // need to rewrite the node with the same index operation
+                    if (const RamIndexScan* iscan = dynamic_cast<RamIndexScan*>(node.get())) {
+                        node = std::make_unique<RamIndexScan>(
+                                std::make_unique<RamRelationReference>(&iscan->getRelation()),
+                                iscan->getTupleId(), std::move(updatedPattern), std::move(filter),
+                                iscan->getProfileText());
+                    } else if (const RamParallelIndexScan* pscan =
+                                       dynamic_cast<RamParallelIndexScan*>(node.get())) {
+                        node = std::make_unique<RamParallelIndexScan>(
+                                std::make_unique<RamRelationReference>(&pscan->getRelation()),
+                                pscan->getTupleId(), std::move(updatedPattern), std::move(filter),
+                                pscan->getProfileText());
+                    } else if (const RamIndexChoice* ichoice = dynamic_cast<RamIndexChoice*>(node.get())) {
+                        node = std::make_unique<RamIndexChoice>(
+                                std::make_unique<RamRelationReference>(&ichoice->getRelation()),
+                                ichoice->getTupleId(),
+                                std::unique_ptr<RamCondition>(ichoice->getCondition().clone()),
+                                std::move(updatedPattern), std::move(filter), ichoice->getProfileText());
+                    } else if (const RamIndexAggregate* iagg = dynamic_cast<RamIndexAggregate*>(node.get())) {
+                        node = std::make_unique<RamIndexAggregate>(std::move(filter), iagg->getFunction(),
+                                std::make_unique<RamRelationReference>(&iagg->getRelation()),
+                                std::unique_ptr<RamExpression>(iagg->getExpression().clone()),
+                                std::unique_ptr<RamCondition>(iagg->getCondition().clone()),
+                                std::move(updatedPattern), iagg->getTupleId());
+                    }
```

Comment:
You need to execute abort()/assert(false) here so that we don't have a silent failure in case we have a new class.

Reply:
`souffle::fatal` is intended to be used instead of abort.